### PR TITLE
Add feature: Logging in with Google Account

### PR DESCRIPTION
**Pull request by ajtran303**
### New Migration
Create users table by running `rails db:migrate` - you may have to do it again for the test environment too.

Users are stored in the database with a name and a user_id that comes from an `auth_hash` in the SessionsController.

### New feature: A user can log in with Google by clicking a button …

The user will be created in the database if they have not been already
When a user is logged in, there is a message at the top of every page

The message will not appear if the user is not logged in.
The absence of the message was tested by deleting the cookies in the local development server and refreshing the page.

### Implementation details:

A session will be created when a visitor successfully logs in
The user information comes from the omniauth gem:

```ruby
  def create
    auth_hash = request.env['omniauth.auth']
  ....
```

The information stored in the session does NOT come from the auth_hash!

```ruby
  session[:user_id] = user.id
  # .id refers to the auto-incrementing primary key from the Users table
```

### CODE REVIEW

File: spec/features/session/create_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require "rails_helper"
```
Comment (Kathybui732) [inline spec/features/session/create_spec.rb:1]: We need to test for any sad paths/edge cases also

File: db/schema.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# you'll amass, the slower it'll run and the greater likelihood for issues).
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema.define(version: 2020_08_19_212714) do
+
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "users", force: :cascade do |t|
```
Comment (Kathybui732) [inline db/schema.rb:18]: Do we add tokens to the database?
